### DISCUSSION TITLE
Should the MSA be performed between (relatively) homologous sequences?

### DISCUSSION THREAD

**Discussion by dongli96** [facebookresearch/esm#127] [Q&A] [answered]
Hi,

Thanks for sharing such an amazing tool. I have a question about the inputs of the MSA transformer (specifically, ESM-MSA-1b). 

It's clear to me that each input (single sample) of the MSA transformer should be a MSA that contains numerous sequences. Now I have a raw dataset of 10k protein sequences. I decide to divide them into 200 sets, i.e. with each set containing 50 sequences, then I run MSA individually for each of the 200 sets. So I will get 200 MSAs if things go well. My question is, should the 50 sequences in each set be homologous? Sequences in my raw dataset have various length (from 20 to 1000). If I shuffle the raw dataset to make sequence distributions of each set similar to each other, it will be very possible that short and long sequences are present in the same set. The MSA software has to "put up with" the long sequences and insert a lot of "-" in the short sequences, which seems unreasonable to me. 

Thanks for your time!

**Comment by tomsercu** [comment, marked as answer]
Yes, your input MSA should be a well-formed MSA with homologous and aligned sequences. See for example hhblits or jackhmmer for industry standard tools to construct such MSAs.
For a set of 10k unrelated protein sequences of varying lengths, i wouldn't expect MSA Transformer to do anything reasonable at all.

**Comment by Xinxinatg** [comment]
is it possible that for each of individual sequence in 10k unrelated ones, a MSA sequences set can be set up, then the MSA ESM could be used to generate embeddings for all of these sequences. Only issue is it require massive amout of work.

**Reply by tomsercu** [reply]
Thanks for pointing that out, correct, if you want each of the 10k sequences to be treated as separate samples, you would need to construct and MSA for each of them. Very reasonable to do depending on which db you construct the MSA against and if you can parallelize it a bit on a compute cluster.
